### PR TITLE
Add `git bigstore fetch <repository>` command

### PR DESCRIPTION
**Pull request by jansel**
`git-bigstore` currently does not work well with a [forking workflow].  In this workflow, there is no central repository to store bigstore metadata/notes.  This patch (and the prior one) adds support for this workflow.  

This patch adds a command:
```
git bigstore fetch upstream
```

Which will take the metadata on the remote `upstream` and merge/push it to the remote `origin`.  One can now manually sync bigstore metadata across forks.  This is still not ideal, but it is an improvement over the current version.

[forking workflow]: https://www.atlassian.com/git/tutorials/comparing-workflows#forking-workflow

### CODE REVIEW

File: bigstore/bigstore.py [hunk 1]
```diff
@@ -183,22 +183,35 @@ def pathnames():
...
+            sys.stderr.write('done\n')
         except git.exc.GitCommandError:
             # If it fails silently, an existing notes object already exists.
-            sys.stderr.write("\n")
+            sys.stderr.write('\n')
     else:
-        g().notes("--ref=bigstore", "merge", "-s", "cat_sort_uniq", "refs/notes/bigstore-remote")
-        sys.stderr.write("done\n")
+        g().notes('--ref=bigstore', 'merge', '-s', 'cat_sort_uniq', 'refs/notes/bigstore-remote')
+        sys.stderr.write('done\n')
```
Comment (dlo) [inline bigstore/bigstore.py]: Could you put the quote changes in a separate PR? If we're going to change it we might as well do it across the board and keep it separate from other changes.

File: bigstore/bigstore.py [hunk 2]
```diff
@@ -183,22 +183,35 @@ def pathnames():
...
 
 
-def push():
-    assert_initialized()
+def pull_metadata(repository='origin'):
+    """
+    Pull metadata from repository and automatically merge it with local metadata
+
+    :param repository: git url or remote
+    """
```
Comment (dlo) [inline bigstore/bigstore.py]: 👏🏻

File: bigstore/bigstore.py [hunk 3]
```diff
@@ -336,6 +341,21 @@ def pull():
             sys.stderr.write('ERROR\n')
 
 
+def fetch(repository):
+    """
+    Pull metadata from a remote repository and merge it with our own.
+
+    :param repository: either a git url or name of a remote
+    """
+    assert repository
```
Comment (dlo) [inline bigstore/bigstore.py]: If checking for existence how about an error message if not provided?

Comment (jansel) [inline bigstore/bigstore.py]: An error message is already provided by argparse, this was just defensive coding.  I'll remove the assert.
```
$ git bigstore fetch
usage: git-bigstore fetch [-h] repository
git-bigstore fetch: error: too few arguments
```

File: bigstore/bigstore.py (hunk 1 above)
Comment (jansel) [inline bigstore/bigstore.py]: Done
